### ISSUE TITLE
Lists are Referable but References don't work

### ISSUE THREAD

**Issue by bmaclach** [JacquesCarette/Drasil#1216]
I tried making a reference to the list of Goal Statements and saw that in the final PDF/HTML, the link did not work. The reference itself was created correctly, but it does nothing because a label for the list was never actually rendered. For example, in TeX, the reference `\hyperref[Lst:goalStmt]{Lst: goalStmt}` is properly generated but there is nowhere with the label `Lst:goalStmt`.

I did a little investigation and I think the issue is that we drop the `Label` information from the `List` when we make a `LayoutObj` out of it. Note how a `List` `LayoutObj` is not parameterized by a `Label`: 
https://github.com/JacquesCarette/Drasil/blob/27cf6da8ac0a0e3058e3e75dc1d4744ef697bad6/code/drasil-printers/Language/Drasil/Printing/LayoutObj.hs#L20-L31

**Comment by Mornix** [comment]
Is the desired reference really to the list of goal statements (specifically) or to the concept of goal statements (i.e. the section)?

**Comment by bmaclach** [comment]
To the list of goal statements specifically. Though what I really want to do is reference individual goal statements, but couldn't (#1215), so I tried this as another option. I suppose referring to the goal statements section would be just as effective, though.

**Comment by JacquesCarette** [comment]
So I don't think we should be able to refer to a list in its entirety? An item in a list, definitely. The section in which the list lives, that too. Of course, on top of #1215 being a bug, I think the current state is a bug but in a different way: a list should not be `Referable` at all.

**Comment by Awurama-N** [comment]
I initially thought I would have to add `Label` as a parameter for lists, from the first comment, but i see you said list should not be referrable at all. Please How could i stop this?
@balacij

**Comment by balacij** [comment]
Hmm, I'm not quite sure if we even have any "Lst:" references at all. I tried to search for it to no avail. Do you have any specific locations they occur in @Awurama-N ? I see some small helper functions in docLang/Drasil.DocumentLanguage.Labels but I don't see any real usage of them. If the goal is to remove the functionality of references to Lists, a part of this will surely be to remove these helper functions (`grep "makeLstRef" -r * --include *.hs`). If the data constructors are completely hidden away, I don't think we would need to do much else other than that really.

Also, from the looks of it, the References/IDs for the items in the list are already working (according to the `makeList` function of both `drasil-printers/Language.Drasil.HTML.Print` & `drasil-printers/Language.Drasil.TeX.Print`). Did you validate this/find any working examples?

@Ant13731 Any thoughts on this?

**Comment by Ant13731** [comment]
Yea, just looking through the examples, I don't see any real use. I think all the functions that use `makeLstRef` could just be deleted, and turn the labelled content in the examples into unlabeled content (I think just switching `LlC` for `UlC` should work, though there might be another function needed, `ulcc` maybe?). 

If we really wanted to refer to a single point in a list, we could make it the point itself `ConceptInstance` like `Assumptions` currently does.

**Comment by balacij** [comment]
Are the references to single items not working already? From looking at the code, it looks like it _should_ work.

**Comment by Ant13731** [comment]
Ohh I see. Yes I think they already do work. For the goal statements, individual goals are already laid out as `ConceptInstances` and that's how they are referred to.

But for something like the user characteristics in GlassBR, wouldn't the section reference be enough? I don't think we reference any of the points individually.

**Comment by balacij** [comment]
Yep, section references are fine, I believe this is what Dr. Carette was discussing earlier. Thanks @Ant13731!

In that case, @Awurama-N, it sounds like you don't need to edit the `List` constructor at all, but you should look around for the aforementioned removable `makeLst`-like functions and remove them, and then this ticket sounds seems complete.

**Comment by Awurama-N** [comment]
okay thank you
